src/plugins/data_source/common/data_connections/data_connection_saved_object_attributes.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectAttributes } from 'src/core/types';
+
+export const DATA_CONNECTION_SAVED_OBJECT_TYPE = 'data-connection';
+
+export interface DataConnectionSavedObjectAttributes extends SavedObjectAttributes {
+  connectionId: string;
```

Comment:
doesn't this really equate to the `title` of the saved object?

Reply:
It will be the id of the particular connect and will not be mutable by nature, hence the term id. There saved object in itself doesn't force the attributes to have a particular title attribute, so we should be okay.

Reply:
yeah should be ok. perhaps it might be beneficial to add some TSDOC in line comments that specify that this `connectionId` is not the saved object connection id but the API specific `connectionId` param. Since the current status quo of the repo has been `id` usually referencing the saved object ID. and title was the actual name. where as with the new services seem to make the `id` concept the name of the connection data structure

Reply:
i know i constantly end up sending the ID instead of the name of the data structure lately due to these conflicting definitions of ID between OpenSearch and other

Reply:
Added the comments.